samples/XAMLCatalog/XAMLCatalog/UIKitControls/UILabelViewController.m

```diff
@@ -14,351 +14,309 @@
-    textLabel.lineBreakMode = lineBreakMode;
-
-    if (minimumFontSize > textLabel.minimumFontSize) {
-        textLabel.minimumFontSize = minimumFontSize;
+- (void)enumerateNumberOfLinesLinBreakFontSize {
+    // enumerate different number of lines
+    for (int numberOfLines = 0; numberOfLines <= 4; numberOfLines++) {
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            self.numberOfLines.text = [NSString stringWithFormat:@"%d", numberOfLines];
+            [self textFieldEditDidEnd:self.numberOfLines];
+        });
+
+        // enumerate different lineBreak
+        for (int lineBreakMode = 0; lineBreakMode <= (int)NSLineBreakByTruncatingMiddle; lineBreakMode++) {
+            dispatch_sync(dispatch_get_main_queue(), ^{
+                self.lineBreakMode.text = [NSString stringWithFormat:@"%d", lineBreakMode];
+                [self textFieldEditDidEnd:self.lineBreakMode];
+            });
+
+            // enumerate font size
+            const float sizeIncreaseStep = 8.00f;
+
+            __block float min;
+            __block float max;
+            dispatch_sync(dispatch_get_main_queue(), ^{
+                min = self.fontSize.minimumValue;
+                max = self.fontSize.maximumValue;
+            });
+
+            for (float size = min; size <= max; size += sizeIncreaseStep) {
+                dispatch_sync(dispatch_get_main_queue(), ^{
+                    self.fontSize.value = size;
+                    [self fontSizeChanged:self.fontSize];
+                    [self updateLog];
+                });
+            }
+        }
     }
+}
 
-    BOOL shouldAdjustSize = adjustFontSizeToFitWidth && (lineBreakMode != UILineBreakModeWordWrap) &&
-                            (lineBreakMode != NSLineBreakByCharWrapping) && (minimumFontSize < 17.0);
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+- (BOOL)textFieldShouldReturn:(UITextField*)textField {
+    [textField resignFirstResponder];
+    return YES;
+}
 
-    textLabel.text = [NSString
-        stringWithFormat:@"adjustFontSize = %d, numberOflines = %d, linbreakMode=%d, text %@ adjust to fit the width of this UIlabel",
-                         textLabel.adjustsFontSizeToFitWidth,
-                         textLabel.numberOfLines,
-                         textLabel.lineBreakMode,
-                         shouldAdjustSize ? @"should" : @"should not"];
+- (void)autoShrinkChanged:(UISwitch*)autoShrinkSwtich {
+    self.minimumFontSize.enabled = self.minimumFontSize.enabled = self.autoShrink.on;
+    self.label.adjustsFontSizeToFitWidth = self.autoShrink.on;
+}
 
-    return textLabel;
+- (void)fontSizeChanged:(UISlider*)slider {
+    self.label.font = [self.label.font fontWithSize:slider.value];
+    self.fontSizeLabel.text = [NSString stringWithFormat:@"%5.2f", slider.value];
+}
+
+- (void)textFieldEditDidEnd:(UITextField*)t {
```

Comment:
sure :)

---
In reply to: [104797653](https://github.com/Microsoft/WinObjC/pull/2178#discussion_r104797653) [](ancestors = 104797653)